packages/shared/src/components/MainFeedLayout.tsx

```diff
@@ -379,6 +365,45 @@ export default function MainFeedLayout({
     }
   }, [sortingEnabled, selectedAlgo, loadedSettings, loadedAlgo]);
 
+  useEffect(() => {
```

Comment:
The following two useEffects are quite big and specific, could we perhaps move them to a custom hook instead?

It will make it more understandable what's happening.

Reply:
Good point. Let me see how we can move out these code blocks that also involve the modal.